device/transport/http/test/_http_test.js

```diff
@@ -349,6 +349,12 @@ describe('Http', function () {
         transport.setOptions({});
       });
     });
+
+    it('sets _productInfo if productInfo is provided', function() {
+      var transport = new Http(fakeAuthenticationProvider);
+      transport.setOptions({ productInfo: 'fakeProductInfoString' });
+      assert.exists(transport._productInfo);
+    });
```

Comment:
a better test would be to use a mock object as the second parameter of the Http constructor and assert that the common HTTP object is called with the User-Agent header that contains productInfo